### PR TITLE
add dbshell tests

### PR DESCRIPTION
**Pull request by WaVEV**
# implement dbshell #3
implement dbshell

ticket-#3

# Checklist
- [x] Unit test for MongoDB shell

### CODE REVIEW

File: tests/dbshell/test_mongodb.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+                        },
+                        ["--help"],
+                    ),
+                    (["mongosh", "somedbname", "--help"], None),
+                )
+
+    def test_crash_password_does_not_leak(self):
+        # The password doesn't leak in an exception that results from a client
+        # crash.
+        self.skipTest("We are unable to pass the password via environment vairable")
```
Comment (timgraham) [inline tests/dbshell/test_mongodb.py]: You can remove this test.

File: tests/dbshell/test_mongodb.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+from pathlib import Path
+from unittest import mock
+
+from django_mongodb.client import DatabaseClient
+
+from django.db import connection
+from django.test import SimpleTestCase
+
+
+class MongoDBDbshellCommandTestCase(SimpleTestCase):
```
Comment (timgraham) [inline tests/dbshell/test_mongodb.py]: I guess you copied this from another example but "CommandTestCase" -> "Tests". The name "TestCase" implies its meant to be inherited from. Some Django tests are a little sloppy about this.

File: tests/dbshell/test_mongodb.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                        "someuser",
+                        "--password",
+                        "somepassword",
+                        "--authenticationDatabase",
+                        "value1",
+                        "--authenticationMechanism",
+                        "value2",
+                        "--retryWrites",
+                        "true",
+                        "mongodb://somehost:444/somedbname",
```
Comment (timgraham) [inline tests/dbshell/test_mongodb.py]: I guess you didn't update this for the latest version which no longer uses URIs.

File: tests/dbshell/test_mongodb.py (hunk 2 above)
Comment (WaVEV) [reply tests/dbshell/test_mongodb.py]: Changed to MongoDBDbshellCommandTests

File: tests/dbshell/test_mongodb.py (hunk 1 above)
Comment (WaVEV) [reply tests/dbshell/test_mongodb.py]: Removed

File: tests/dbshell/test_mongodb.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                                "authenticationDatabase": authentication_database,
+                                "authenticationMechanism": authentication_mechanism,
+                                "retryWrites": retry_writes,
+                            },
+                        }
+                    ),
+                    (expected_args, None),
+                )
+
+    def test_options_override_settings_proper_values(self):
```
Comment (timgraham) [inline tests/dbshell/test_mongodb.py:73]: This test isn't needed. It's for a line in MySQL's client.py: 
```
port = settings_dict["OPTIONS"].get("port", settings_dict["PORT"])
```
MongoDB doesn't have that.

**Review by timgraham — APPROVED** [review]
Squash commits with commit message: "add dbshell tests"

**Pull request merged**